vault/external_tests/pprof/pprof_test.go

```diff
@@ -194,9 +226,36 @@ func TestSysPprof_Standby(t *testing.T) {
 			},
 		},
 	})
-	cluster.Start()
 	defer cluster.Cleanup()
 
+	testSysPprof_Standby(t, cluster)
+}
+
+// TestSysPprof_Standby_Exec is the same as TestSysPprof_Standby, but using a Vault binary
+// running as -dev-three-node instead of a fake single node TestCluster.  There's
+// no particular reason why TestSysPprof was chosen to validate that mechanism,
+// other than that it was fast and simple.
+func TestSysPprof_Standby_Exec(t *testing.T) {
+	t.Parallel()
+	binary := os.Getenv("VAULT_BINARY")
+	if binary == "" {
+		t.Skip("only running exec test when $VAULT_BINARY present")
+	}
+	cluster := testcluster.NewTestExecDevCluster(t, &testcluster.ExecDevClusterOptions{
+		ClusterOptions: testcluster.ClusterOptions{
+			VaultNodeConfig: &testcluster.VaultNodeConfig{
+				DisablePerformanceStandby: true,
+			},
+		},
+		BinaryPath:        binary,
```

Comment:
What's your thoughts about an `auto` value? It'd be cool to just set this globally and have it automatically find it from the `bin/vault` of the repo that its running tests from. 

Then I can set it from the base shell profile and have it magically work across OSS/ENT/... -- worst case the binary is out of date if I haven't rebuilt it before running tests, but at least they'll run by default more often (and CI will obviously pick up if there's an unexpected local test pass due to mismatched binary version).

This lets the empty string also still mean "don't run these tests".

Reply:
Note that typically the bin/vault isn't going to be the right GOOS (and possibly not the right GOARCH) for use in docker.

I'm not averse to us taking steps to run these tests automatically locally, but I'm inclined to save that for a future iteration.  Ideally I'd want it to actually build the binary if missing/stale, though maybe that's something we could instead delegate to the IDE.

Reply:
Ahh I see. Hmm... seems like we'd need to add `$GOOS`/`$GOARCH` into the bin path somehow, but that does end up being more work in our build scripts. A future problem then!